internal/auth/repository/eventsourcing/eventstore/user.go

```diff
@@ -45,7 +46,40 @@ func (repo *UserRepo) Register(ctx context.Context, registerUser *model.User, or
 	if err != nil {
 		return nil, err
 	}
-	user, aggregates, err := repo.UserEvents.PrepareRegisterUser(ctx, registerUser, pwPolicy, orgPolicy, resourceOwner)
+	user, aggregates, err := repo.UserEvents.PrepareRegisterUser(ctx, registerUser, nil, pwPolicy, orgPolicy, resourceOwner)
```

Comment:
Register and RegisterExternalUser do exactly the same thing with the difference of Register not taking ExternalIDP as param... i'd expose Register as separate func and either call RegisterExternalUser with externalIDP set to nil or create a private func with both of them use